pumpkin-solver/src/propagators/arithmetic/linear_less_or_equal.rs

```diff
@@ -41,57 +74,97 @@ where
             );
         });
 
-        Ok(())
+        self.recalculate_incremental_state(context.as_readonly());
+
+        if let Some(conjunction) = self.detect_inconsistency(context.as_readonly()) {
+            Err(conjunction)
+        } else {
+            Ok(())
+        }
     }
 
-    fn priority(&self) -> u32 {
-        0
+    fn detect_inconsistency(
+        &self,
+        context: PropagationContext,
+    ) -> Option<PropositionalConjunction> {
+        if self.c < self.lower_bound_left_hand_side {
+            let reason: PropositionalConjunction = self
+                .x
+                .iter()
+                .map(|var| predicate![var >= context.lower_bound(var)])
+                .collect();
+            Some(reason)
+        } else {
+            None
+        }
     }
 
-    fn name(&self) -> &str {
-        "LinearLeq"
+    fn notify(
+        &mut self,
+        context: PropagationContext,
+        local_id: LocalId,
+        _event: OpaqueDomainEvent,
+    ) -> EnqueueDecision {
+        let index = local_id.unpack() as usize;
+
+        let x_i = &self.x[index];
+        let old_bound = self.current_bounds[index];
+        let new_bound = context.lower_bound(x_i);
+
+        pumpkin_assert_simple!(
+            old_bound < new_bound,
+            "propagator should only be triggered when lower bounds are tightened, old_bound={old_bound}, new_bound={new_bound}"
+        );
+
+        self.current_bounds[index] = new_bound;
+        self.lower_bound_left_hand_side += new_bound - old_bound;
+
+        EnqueueDecision::Enqueue
     }
 
-    fn debug_propagate_from_scratch(&self, context: PropagationContextMut) -> PropagationStatusCP {
-        perform_propagation(context, &self.x, self.c)
+    fn synchronise(&mut self, context: PropagationContext) {
+        self.recalculate_incremental_state(context);
```

Comment:
If we do want to commit to using the `synchronise` rather than the backtrack events; I noticed that for the cumulative it was much more efficient to store a flag which indicated that a recalculation should take place rather than recalculating eagerly. Would that make sense here?